src/Symfony/Component/Console/Descriptor/TextDescriptor.php

```diff
@@ -198,16 +198,25 @@ protected function describeApplication(Application $application, array $options
             }
 
             // add commands by namespace
+            $commandNames = [];
+            foreach($description->getCommands() as $name => $command) {
+                $commandNames[] = $name;
```

Comment:
I suggest removing this foreach loop and replace it with `$commands = $description->getCommands();`